### ISSUE TITLE
Explicitly allow or disallow use of Entity Manager with extended Persistence Context for CDI injection

### ISSUE THREAD

**Issue by lukasj** [jakartaee/persistence#46]
CDI specification has so called producer methods, allowing to create (or retrieve) objects to be injected.

JPA specification should explicitly state whether is it allowed for Stateful Session bean to have a producer method that returns extended EntityManager?
For example, is this legal:

```java
@ConversationScoped
@Stateful
public class MyConversationBean {
@PersistenceContext(type=EXTENDED)
private EntityManager em;

@Produces
public EntityManager getEntityManager()

{ return em; }

}
```

This would be useful if conversation is implemented with several session beans and they all want to share a single extended EntityManager.

**Comment by m-reza-rahman** [comment]
The extended persistence context never really took off and is now very out of favor. I would just close this to reduce clutter.

Reza Rahman
Jakarta EE Ambassador, Author, Blogger, Speaker

Please note views expressed here are my own as an individual community member and do not reflect the views of my employer.

**Comment by gavinking** [comment]
> JPA specification should explicitly state whether is it allowed for Stateful Session bean to have a producer method that returns extended `EntityManager`?

After so many years, I don't remember precisely what we said about stateful session beans, or where we said it. (Does anybody session beans anymore??!) 

However, I believe the functionality described here was always required, right from JPA 1.0/EJB 3.0, but it was defined by the EJB spec, not by the JPA spec.

(And it's certainly the case that CDI lets you bind an extended persistence context to the conversation scope.)

As far as I can see, it's OK if the JPA spec stays silent on these issues.

**Comment by gavinking** [comment]
(Actually there's some ambiguity here. The title of the issue refers to CDI, but then the code example uses `@PersistenceContext` to do the injection instead of `@Inject`. So I'm not even sure precisely what it is that the issue thinks the JPA spec leaves undefined. Regardless, it's not the job of the JPA spec to define this, AFAICT.)

**Comment by gavinking** [comment]
So I think this is just a dupe of #3 and maybe #377, and so to the extent that there was ever a real "issue" here, it's fixed by #460.
